pkg/cgroups/cgroups_supported.go

```diff
@@ -96,6 +97,17 @@ func UserOwnsCurrentSystemdCgroup() (bool, error) {
 // It differs from os.RemoveAll as it doesn't attempt to unlink files.
 // On cgroupfs we are allowed only to rmdir empty directories.
 func rmDirRecursively(path string) error {
+	killProcesses := func(signal syscall.Signal) {
+		// kill all the processes that are still part of the cgroup
+		if procs, err := ioutil.ReadFile(filepath.Join(path, "cgroup.procs")); err == nil {
```

Comment:
Could we utitlise `cgroup.kill` if available

Reply:
we could for SIGKILL, but we will still need the code for SIGTERM

Reply:
amended the change

Reply:
Thanks looks good to me, do we need to check if `kernel` is above `5.14+` before using `cgroup.kill`